### PR TITLE
ci: setup CODEOWNERS for ml-sys team.

### PR DESCRIPTION
**Pull request by ioga**
## Description

Setup codeowners for ml-sys areas:
- harness except for autogen
- expconf
- python bindings generator
- select e2e_tests (e.g. `experiment` (core UX tests) or `command` (mostly CLI tests), but not `cluster`)

As a next step, we'll need to:

- check if we like the assignment process: https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team#about-auto-assignment
- TBD: enable branch protection
- iterate: if other high-traffic or auto-generated files will surface up, remove them.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by dannysauer** [comment]
This has some overlap with #7545.  Specifically, @aciborowska added `/cli` (and was less specific about harness).

**Comment by dannysauer** [comment]
Whoops.  @azhou-determined.  Dang alphabet. :)

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -9,3 +9,18 @@
...
+/harness  @determined-ai/ml-sys
+/harness/determined/deploy
+/harness/determined/cli
+/harness/tests/cli
+/harness/determined/common/api/bindings.py
+/harness/determined/__version__.py
+
+/schemas                          @determined-ai/ml-sys
+/bindings/generate_bindings_py.py @determined-ai/ml-sys
+/e2e_tests/tests/*.py             @determined-ai/ml-sys
```
Comment (dannysauer) [inline .github/CODEOWNERS:24]: Is tests/requirements.txt also ml-sys?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -9,3 +9,19 @@
...
```
Comment (wes-turner) [inline .github/CODEOWNERS:21]: `e2_tests` is in a rough place. I think it needs an owner, and I support your suggestion that that owner be ml-sys. If it does turn out to be ml-sys, we should schedule some time in the next quarter to clean it up and add a dev guide for it.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -9,3 +9,19 @@
 /master   @determined-ai/backend
 /agent    @determined-ai/backend
 /proto    @determined-ai/backend
+
+# Harness is mapped to ml-sys by default, except for `cli`, `deploy`,
+# and auto-generated bindings and version files.
+/harness  @determined-ai/ml-sys
+/harness/determined/deploy
+/harness/determined/cli
```
Comment (wes-turner) [inline .github/CODEOWNERS]: Echoing Slack, I think ml-sys is also a good owner for CLI (code and tests) and det deploy.

**Review by wes-turner — APPROVED** [review]

File: .github/CODEOWNERS (hunk 3 above)
Comment (wes-turner) [reply .github/CODEOWNERS]: already done

File: .github/CODEOWNERS (hunk 2 above)
Comment (mpkouznetsov) [reply .github/CODEOWNERS:21]: @MikhailKardash is looking into pruning the examples, which might be related to cleaning up `e2e_tests`.

**Review by mpkouznetsov — APPROVED** [review]
LGTM as a first step.

**Review by azhou-determined — APPROVED** [review]

**Comment by azhou-determined** [comment]
I think cli should belong to ml-sys. It's a user-facing API that interacts with the backend server, and it's python code.

**Review by dannysauer — APPROVED** [review]
I'm also good with this as-is, though it doesn't address `/cli` from #7545.

**Comment by ioga** [comment]
`/cli` directory does not exist in this repo. `/harness/determined/cli/` is the proper path, this PR should have it covered.

**Pull request merged**